contrib/net/http/trace.go

```diff
@@ -41,20 +42,33 @@ type ServeConfig struct {
 // TraceAndServe serves the handler h using the given ResponseWriter and Request, applying tracing
 // according to the specified config.
 func TraceAndServe(h http.Handler, w http.ResponseWriter, r *http.Request, cfg *ServeConfig) {
-	if cfg == nil {
-		cfg = new(ServeConfig)
+	span, ctx := StartRequestSpan(r, cfg.Service, cfg.Resource, cfg.QueryParams, cfg.SpanOpts...)
+	rw, ddrw := wrapResponseWriter(w)
+	defer func() {
+		FinishRequestSpan(span, ddrw.status, cfg.FinishOpts...)
+	}()
+
+	if appsec.Enabled() {
+		h = httpsec.WrapHandler(h, span, cfg.RouteParams)
 	}
+	h.ServeHTTP(rw, r.WithContext(ctx))
+}
+
+// StartRequestSpan starts an HTTP request span with the standard list of HTTP request span tags. URL query parameters
+// are added to the URL tag when queryParams is true. Any further span start option can be added with opts.
+func StartRequestSpan(r *http.Request, service, resource string, queryParams bool, opts ...ddtrace.StartSpanOption) (tracer.Span, context.Context) {
 	path := r.URL.Path
-	if cfg.QueryParams {
+	if queryParams {
 		path += "?" + r.URL.RawQuery
 	}
-	opts := append([]ddtrace.StartSpanOption{
+	opts = append([]ddtrace.StartSpanOption{
```

Comment:
Should we always add `tracer.Measured()` as it is almost always present in the contribs?

Reply:
Same question for `FinishRequestSpan`. That sounds like something that shouldn't be part of the API.

Reply:
Now moved into an internal package in https://github.com/DataDog/dd-trace-go/pull/1286/commits/029b8b5b8f134c893550392a960ffbb4a10ac264